google-cloud-spanner/acceptance/spanner/client/types/date_test.rb

```diff
@@ -15,134 +15,153 @@
-    db.upsert table_name, { id: id, dates: nil }
-    results = db.read table_name, [:id, :dates], keys: id
-
-    _(results).must_be_kind_of Google::Cloud::Spanner::Results
-    _(results.fields.to_h).must_equal({ id: :INT64, dates: [:DATE] })
-    _(results.rows.first.to_h).must_equal({ id: id, dates: nil })
-  end
-
-  it "writes and queries NULL array of date" do
-    id = SecureRandom.int64
-    db.upsert table_name, { id: id, dates: nil }
-    results = db.execute_query "SELECT id, dates FROM #{table_name} WHERE id = @id", params: { id: id }
-
-    _(results).must_be_kind_of Google::Cloud::Spanner::Results
-    _(results.fields.to_h).must_equal({ id: :INT64, dates: [:DATE] })
-    _(results.rows.first.to_h).must_equal({ id: id, dates: nil })
+  dialects = [:gsql]
+  dialects.push :pg unless emulator_enabled?
+
+  dialects.each do |dialect|
+    it "writes and reads date #{dialect}" do
+      id = SecureRandom.int64
+      db[dialect].upsert table_name, { id: id, date: Date.parse("2017-01-01") }
+      results = db[dialect].read table_name, [:id, :date], keys: id
+
+      _(results).must_be_kind_of Google::Cloud::Spanner::Results
+      _(results.fields.to_h).must_equal({ id: :INT64, date: :DATE })
+      _(results.rows.first.to_h).must_equal({ id: id, date: Date.parse("2017-01-01") })
+    end
+
+    it "writes and queries date #{dialect}" do
+      id = SecureRandom.int64
+      db[dialect].upsert table_name, { id: id, date: Date.parse("2017-01-01") }
+      results = db[dialect].execute_query date_query[dialect], params: dialect === :gsql ? { id: id } : { p1: id }
+
+      _(results).must_be_kind_of Google::Cloud::Spanner::Results
+      _(results.fields.to_h).must_equal({ id: :INT64, date: :DATE })
+      _(results.rows.first.to_h).must_equal({ id: id, date: Date.parse("2017-01-01") })
+    end
+
+    it "writes and reads NULL date #{dialect}" do
+      id = SecureRandom.int64
+      db[dialect].upsert table_name, { id: id, date: nil }
+      results = db[dialect].read table_name, [:id, :date], keys: id
+
+      _(results).must_be_kind_of Google::Cloud::Spanner::Results
+      _(results.fields.to_h).must_equal({ id: :INT64, date: :DATE })
+      _(results.rows.first.to_h).must_equal({ id: id, date: nil })
+    end
+
+    it "writes and queries NULL date #{dialect}" do
+      id = SecureRandom.int64
+      db[dialect].upsert table_name, { id: id, date: nil }
+      results = db[dialect].execute_query date_query[dialect], params: dialect === :gsql ? { id: id } : { p1: id }
+
+      _(results).must_be_kind_of Google::Cloud::Spanner::Results
+      _(results.fields.to_h).must_equal({ id: :INT64, date: :DATE })
+      _(results.rows.first.to_h).must_equal({ id: id, date: nil })
+    end
+
+    it "writes and reads array of date #{dialect}" do
+      id = SecureRandom.int64
+      db[dialect].upsert table_name,
+                         { id: id,
+dates: [Date.parse("2016-12-30"), Date.parse("2016-12-31"), Date.parse("2017-01-01")] }
```

Comment:
Please watch the indentation. Here, and in several additional places below. Use rubocop if you think it will help.

Reply:
Actually rubocop's autocorrect was messing up the indentation, will try checking later if there is a config to fix this